dotnet/src/Extensions/PromptTemplates.Liquid/LiquidPromptTemplate.cs

```diff
@@ -16,22 +17,29 @@ namespace Microsoft.SemanticKernel.PromptTemplates.Liquid;
 /// </summary>
 internal sealed class LiquidPromptTemplate : IPromptTemplate
 {
+    private const string ReservedString = "&#58;";
+    private const string ColonString = ":";
+    private readonly PromptTemplateConfig _config;
+    private readonly bool _allowUnsafeContent;
     private static readonly Regex s_roleRegex = new(@"(?<role>system|assistant|user|function):\s+", RegexOptions.Compiled);
 
     private readonly Template _liquidTemplate;
     private readonly Dictionary<string, object> _inputVariables;
 
     /// <summary>Initializes the <see cref="LiquidPromptTemplate"/>.</summary>
     /// <param name="config">Prompt template configuration</param>
-    /// <exception cref="ArgumentException"><see cref="PromptTemplateConfig.TemplateFormat"/> is not <see cref="LiquidPromptTemplateFactory.LiquidTemplateFormat"/>.</exception>
+    /// <param name="allowUnsafeContent">Whether to allow unsafe content in the template</param>
+    /// <exception cref="ArgumentException">throw if <see cref="PromptTemplateConfig.TemplateFormat"/> is not <see cref="LiquidPromptTemplateFactory.LiquidTemplateFormat"/></exception>
     /// <exception cref="ArgumentException">The template in <paramref name="config"/> could not be parsed.</exception>
-    public LiquidPromptTemplate(PromptTemplateConfig config)
+    public LiquidPromptTemplate(PromptTemplateConfig config, bool allowUnsafeContent = false)
```

Comment:
We should add logging, this can come later

Reply:
Track in https://github.com/microsoft/semantic-kernel/issues/6033